pkg/ipam/poolallocator/allocator.go

```diff
@@ -252,10 +260,12 @@ func (a *IPPoolAllocator) removeIPAddressState(ipPool *v1alpha2.IPPool, ip net.I
 // allocated, or in case CRD failed to update its state.
 // In case of success, IP pool CRD status is updated with allocated IP/state/resource/container.
 // AllocateIP returns subnet details for the requested IP, as defined in IP pool spec.
-func (a *IPPoolAllocator) AllocateIP(ip net.IP, state v1alpha2.IPAddressPhase, owner v1alpha2.IPAddressOwner) (v1alpha2.SubnetInfo, error) {
+func (a *IPPoolAllocator) AllocateIP(ip net.IP, state v1alpha2.IPAddressPhase, owner v1alpha2.IPAddressOwner) (v1alpha2.SubnetInfo, *uint32, error) {
```

Comment:
Should we define a new type for IP allocation result, or at least a new type for SubnetInfo, which uses net.IP/net.IPNet, and uint16 for VLAN ID?

Reply:
My miss. `SubnetInfo` already contains VLAN, removed *uint32 from the return list.

Reply:
But that is a string. I am suggesting to define a new type for internal uses, which should use net.IP/IPNet and uint16.

Reply:
Got your point.
I think current implementation by Anna performs well, and I don't have strong wish to move this to internal struct.
@annakhm Could you explain why we prefer to use `v1alpha2.SubnetInfo` instead of a internal struct which contains parsed data such as net.IP?

Reply:
I think we should use Golang types rather than string, if no good reason not doing that.

BTW, why VLAN in SubnetInfo is a string but not uint16?

Reply:
It is to support VLAN auto allocation in future.
@annakhm Could you explain this more?

Reply:
Originally the idea was to provide the user with configured information (GW, vlan..). Parsed info might not be needed by the user. But I don't see a problem changing that if needed.

Reply:
@annakhm : why SubnetInfo in the CRD definition uses a string but not uint16 for VLAN ID? (BTW, we might use uint16 but not int32 for PrefixLength too?).

Reply:
The idea was to allow to configure wildcard VLAN that would trigger auto-assigning VLAN.

Reply:
Chatted with @annakhm offline. We agree better to use integer for VLAN in SubnetInfo. I think we can make the change in a separate PR.